src/algebra/quandle.lean

```diff
@@ -89,6 +90,14 @@ class shelf (α : Type u) :=
 (act : α → α → α)
 (self_distrib : ∀ {x y z : α}, act x (act y z) = act (act x y) (act x z))
 
+/--
+A *unital shelf* is a shelf equipped with an element `1` such that,
+ for all elements `x`, we have both `x ◃ 1` and `1 ◃ x` equal `x`.
+-/
+class unital_shelf (α : Type u) extends shelf α, has_one α :=
+(one_act : ∀ {a : α}, act 1 a = a)
+(act_one : ∀ {a : α}, act a 1 = a)
```

Comment:
With this design, I fear that then we might need `unital_rack` and `unital_quandle` as well. An alternative is to create what are being called "mixin" classes.
```lean
class is_unital_shelf (α : Type u) [shelf α] [has_one α] : Prop :=
(one_act : ∀ {a : α}, act 1 a = a)
(act_one : ∀ {a : α}, act a 1 = a)
```
Then when you introduce, for example, a unital quandle, you'd do `variables (α : Type*) [quandle α] [has_one α] [is_unital_shelf α]`.

This also depends on whether unital quandles and racks are interesting. The lemma `rack.assoc_iff_id` I suppose implies that they're not!

So, let's keep `unital_shelf` as-is but add a comment to the `unital_shelf` docstring saying that `unital_rack` would be trivial due to `rack.assoc_iff_id` and `unital_shelf.assoc`.

(Pinging @eric-wieser who might have thoughts about algebraic hierarchy design.)